### ISSUE TITLE
Open Parties and Claims Integration

### ISSUE THREAD

**Issue by Aeases** [EtherealOnyx/PartiesMod#4]
Integration to have a party from Open-PAC and PartiesMod apply in both mods.

https://www.curseforge.com/minecraft/mc-mods/open-parties-and-claims/files

I'm already using Open-PAC for claims, XP share sounds nice but don't want to have to deal with two different party mods.

**Comment by EtherealOnyx** [comment]
Hi there,

Perhaps having the ability to sync the parties would suffice?
I could implement the ability to send a party invite to all members of your current Open-PAC party in one go, along a server config option to force Open-PAC parties to be in sync (the /party invite, leave, kick, disband command won't work anymore and the party would only be updated when someone joins or leaves your current Open-PAC party).

**Comment by Aeases** [comment]
Would it be possible to have the server config allow u to choose which way to sync, e.g.
```
// Integration with Open-PAC Parties;
// "none" - Ignore other party systems
// "open-pac" - Rely on Open-PAC Parties, disable Parties Mod commands and use Open-PAC parties for P.m features
// "sync" - Rely on Parties Mod Parties, and sync Parties Mod parties to Open-PAC automatically
open-pac-integration:  none
```
<p align=center width=25% >Should be able to sync Parties mod parties to Open-PAC with the <a href="https://thexaero.github.io/open-parties-and-claims/javadoc/xaero/pac/common/server/parties/party/api/IServerPartyAPI.html">api in the image</p>
<a href="https://thexaero.github.io/open-parties-and-claims/javadoc/xaero/pac/common/server/parties/party/api/IServerPartyAPI.html"><img src="https://user-images.githubusercontent.com/96164686/232743773-6997a0b7-c410-416c-93a2-8ebdf6eb61db.png" align=right width="100%" height="75%">

**Comment by EtherealOnyx** [comment]
Sounds like a good idea. Thank you for going out of your way to look for that information, it makes things a lot easier.

I would also like to eventually add support for FTB Teams and perhaps it can use this same config option as well, with another value like `"ftb-teams"`. With that in mind, the configuration name could be more generic like `partiesSync` or `partiesModIntegration`.

Regardless, some variation of this will be implemented in the next version. Thanks!

**Comment by EtherealOnyx** [comment]
Small update after a very long delay (I apologize),

I have already implemented the configuration, I just needed to work on adding party persistence since that is a requirement for parties being synced. If not, the parties would reset on server restart which would probably be very bad. Nevertheless, this has finally been implemented.

I've been working on a few bugs - one of which was quite important to fix (fake-player tile entities can possibly crash). With those done, however, I can finish the integration between these mods as this is now the highest priority.

**Comment by EtherealOnyx** [comment]
I ended up implementing it in a different way since I did not want to modify open PAC functionality - I don't even think that's allowed, to be honest.
The config is now called `partiesSync` which can be `true` or `false`. If false, the parties mod will act like it does now. If true however, it will attempt to sync parties to the ones existing on Open PAC and will modify the Parties Mod parties accordingly based on command usage. It also changes the behavior of the following commands:
`/party invite playerName` - This functions normally, but whenever they accept an invite, it will act as if you invited them to an Open PAC party and add the member on that party (and this mod will sync it accordingly). It will also create a party (`openpac-parties create`) if the inviter did not have a party first.
`/party leader playerName` - This will function the same way as if you transfered ownership through Open PAC. My mod will then sync the leader transfer accordingly.
`/party kick playerName` - This will function the same way as if you used Open PAC's kick command. However, it will only work if the command initiator is the owner (for now). 
`/party leave` - This will act the same as it is now, but will transfer ownership if they're the current leader and the party would not disband.

All the Open-PAC commands should be working as intended as well, since my mod only syncs the parties based on what changes in the other mod. `partiesSync` is disabled by default.

The goal here was that both sets of commands will work as normal, but party modification will prioritize Open-PAC implementations while my mod follows up accordingly.

I have two small (unrelated) issues I need to fix and then I will release the next version with these changes.
